lib/qesapdeployment.pm

```diff
@@ -944,6 +954,351 @@ sub qesap_az_get_peering_name {
+
+=item B<TIMEOUT> - default is 5 mins
+
+=back
+=cut
+
+sub qesap_aws_delete_transit_gateway_vpc_attachment {
+    my (%args) = @_;
+    croak 'Missing mandatory name argument' unless $args{name};
+    $args{timeout} //= bmwqemu::scale_timeout(300);
+
+    my $res = qesap_aws_get_transit_gateway_vpc_attachment(
+        name => $args{name});
+    # Here [0] suppose that only one of them match 'name'
+    my $transit_gateway_attachment_id = $res->[0]->{TransitGatewayAttachmentId};
+    return 0 unless $transit_gateway_attachment_id;
+
+    my $cmd = join(' ', 'aws ec2 delete-transit-gateway-vpc-attachment',
+        '--transit-gateway-attachment-id', $transit_gateway_attachment_id);
+    script_run($cmd);
+
+    my $state = 'none';
+    my $duration;
+    my $start_time = time();
+    while ((($duration = time() - $start_time) < $args{timeout}) && ($state !~ m/deleted/)) {
+        sleep 5;
+        $res = qesap_aws_get_transit_gateway_vpc_attachment(
+            transit_gateway_attach_id => $transit_gateway_attachment_id);
+        $state = $res->[0]->{State};
+    }
+    return $duration < $args{timeout};
+}
+
+=head3 qesap_aws_add_route_to_tgw
+    Adding the route to the TGW to the routing table in refhost VPC
+
+=over 3
+
+=item B<RTABLE_ID> - Routing table ID
+
+=item B<TARGET_IP_NET> - Target IP network to be added to the Routing table eg. 192.168.11.0/16
+
+=item B<TRANSIT_GATEWAY_ID> - ID of the target Transit gateway (IBS Mirror)
+
+=back
+=cut
+
+sub qesap_aws_add_route_to_tgw {
+    my (%args) = @_;
+    croak 'Missing mandatory rtable_id argument' unless $args{rtable_id};
+    croak 'Missing mandatory target_ip_net argument' unless $args{target_ip_net};
+    croak 'Missing mandatory trans_gw_id argument' unless $args{trans_gw_id};
+
+    my $cmd = join(' ',
+        'aws ec2 create-route',
+        '--route-table-id', $args{rtable_id},
+        '--destination-cidr-block', $args{target_ip_net},
+        '--transit-gateway-id', $args{trans_gw_id},
+        '--output text');
+    script_run($cmd);
+}
+
+=head3 qesap_aws_get_mirror_tg
+
+    Return the Transient Gateway ID of the IBS Mirror
+
+=cut
+
+sub qesap_aws_get_mirror_tg {
+    # Get Transit Gateway ID
+    my $cmd = join(' ', 'aws ec2 describe-transit-gateways',
+        '--filters "Name=tag-key,Values=Project" "Name=tag-value,Values=IBS Mirror"',
+        '--query "TransitGateways[].TransitGatewayId"',
+        '--output text');
+    return script_output($cmd);
+}
+
+=head3 qesap_aws_get_vpc_workspace
+
+    Get the VPC tag Workspace
+
+=over 1
+
+=item B<VPC_ID> - VPC ID of resource to be attached (SUT HANA cluster)
+
+=back
+=cut
+
+sub qesap_aws_get_vpc_workspace {
+    my (%args) = @_;
+    croak 'Missing mandatory vpc_id argument' unless $args{vpc_id};
+
+    # Get the VPC tag Workspace
+    my $vpc_filter = "--filters \"Name=vpc-id,Values=$args{vpc_id}\"";
+    my $cmd = join(' ', 'aws ec2 describe-vpcs',
+        $vpc_filter,
```

Comment:
Same question as above. I don't see the need for `$vpc_filter`. I would simply do:
```
    my $cmd = join(' ', 'aws ec2 describe-vpcs',
        "--filters \"Name=vpc-id,Values=$args{vpc_id}\"",
        ...
```